maingate/src/instructions.rs

```diff
@@ -30,69 +28,22 @@ use crate::{AssignedCondition, AssignedValue};
 /// `... + assigned_or_unassigned_witness_0 * fixed_0 +
 /// assigned_or_unassigned_witness_1 * fixed_1 + ... `
 #[derive(Clone)]
-pub enum Term<F: FieldExt> {
+pub enum Term<'a, F: FieldExt> {
     /// Assigned value and fixed scalar
-    Assigned(AssignedValue<F>, F),
+    Assigned(&'a AssignedValue<F>, F),
     /// Unassigned witness and fixed scalar
     Unassigned(Value<F>, F),
     /// Empty term
     Zero,
 }
 
-impl<F: FieldExt> Term<F> {
-    /// Given constant `WIDTH` returns all zero terms
-    pub(crate) fn empty<const WIDTH: usize>() -> [Self; WIDTH] {
-        vec![Self::Zero; WIDTH].try_into().unwrap()
-    }
-
+impl<'a, F: FieldExt> Term<'a, F> {
```

Comment:
The function below(which is not edited in this PR) can be `const`. We should change it.